explore/src/main/scala/explore/constraints/ConstraintsSummaryTable.scala

```diff
@@ -73,10 +76,20 @@ object ConstraintsSummaryTable {
     "observations" -> "Observations"
   )
 
-  private val columnClasses: Map[String, Css]  = Map(
+  private val columnClasses: Map[String, Css]                                             = Map(
     "edit" -> (ExploreStyles.Sticky |+| ExploreStyles.ConstraintsSummaryEdit)
   )
 
+  private def toSortingRules(tuples: List[(String, Boolean)])                             = tuples.map { case (id, b) =>
+    SortingRule[ConstraintGroup](id).setDesc(b)
+  }.toJSArray
+
+  private def fromTableState(state: TableState[ConstraintGroup]): List[(String, Boolean)] = state
+    .asInstanceOf[ConstraintsTable.StateType]
```

Comment:
I would expect things to work if we define `state: ConstraintsTable.StateType` and avoid the cast.

Reply:
Except that `tableInstance.state` returns a `TableState` (no sorting state mixed in). I don't know of another way to obtain the state, but I may be missing something.

Reply:
Ah, right `TableInstance` facade is generated by ScalablyTyped and is not type-parametrized beyond `D`. Sorry, got mixed up.